modules/tensor_mechanics/include/materials/abaqus/utility/DROTCalc.f

```diff
@@ -0,0 +1,378 @@
+      DIMENSION DSTRAN(NTENS)
+      DIMENSION DFGRD0(3,3),DFGRD1(3,3),DROT(3,3),DF(3,3),F0I(3,3)
+      DIMENSION DQDQ(3,3),DFM(3,3),DFP(3,3)
+      DIMENSION VL(3,3),DW(3,3),DM(3,3),DP(3,3)
+      DIMENSION TDR(3,3),TI(3,3)
+C.4-----  Real,Integer,Complex,Double precision,Logical,Character
+C.5-----  Common
+C.6-----  Equivalence
+C.7-----  Save,Data
+      DATA HALF,ONE,TWO/0.5D0,1.D0,2.D0/
+C.8-----  Functions Definition
+C----------------------------------------------------------------------+
+C======================================================================+
+C----------------------------------------------------------------------+
+C-------  Deformation Gradient Increment DF
+C
+      CALL AFFECT(N*N,DFGRD0,F0I)
+      CALL INVERS(F0I,N,N,DET)
+      CALL PRDMAT(N,DFGRD1,F0I,DF)
+C----------------------------------------------------------------------+
+C----------------------------------------------------------------------+
+C-------  Velocity Gradient L(t+Dt/2)
+C
+      DO I=1,N
+         DO J=1,N
+            DFM(I,J) = DF(I,J)
+            DFP(I,J) = DF(I,J)
+         END DO
+         DFM(I,I) = DFM(I,I)-ONE
+         DFP(I,I) = DFP(I,I)+ONE
+      END DO
+C-------
+      CALL INVERS(DFP,N,N,DET)
+      CALL PRDMAT(N,DFM,DFP,VL)
+C-------
+      TWODT = TWO/DTIME
+      DO I=1,N
+         DO J=1,N
+            VL(I,J) = TWODT*VL(I,J)
+         END DO
+      END DO
+C----------------------------------------------------------------------+
+C-------  Strain Increment Tensor Dstran and Plastic Spin DW
+C
+C      DSTRAN(1) = VL(1,1)
+C      DSTRAN(2) = VL(2,2)
+C      DSTRAN(3) = VL(3,3)
+C      DSTRAN(4) = VL(1,2)+VL(2,1)
+C      IF(NTENS.GT.4) THEN
+C         DSTRAN(5) = VL(1,3)+VL(3,1)
+C         DSTRAN(6) = VL(2,3)+VL(3,2)
+C      END IF
+C-------
+      DO I=1,N
+         DO J=1,N
+            DW(I,J) = HALF*(VL(I,J)-VL(J,I))
+         END DO
+      END DO
+C----------------------------------------------------------------------+
+C-------  Increment Rotation Matrix DROT (Jaumann)
+C
+      HALFDT = HALF*DTIME
+      DO I=1,N
+         DO J=1,N
+            DP(I,J) = HALFDT*DW(I,J)
+            DM(I,J) = -DP(I,J)
+         END DO
+         DP(I,I) = DP(I,I)+ONE
+         DM(I,I) = DM(I,I)+ONE
+      END DO
+C-------
+      CALL INVERS(DM,N,N,DET)
+      CALL PRDMAT(N,DM,DP,DROT)
+C----------------------------------------------------------------------+
+C----------------------------------------------------------------------+
+C-------  Comment Check Code
+C
+C      IF(M.GE.4) THEN
+C         WRITE(MPI,*)'WINGET DSTRAN'
+C         DO I=1,NTENS
+C            WRITE(MPI,101)DSTRAN(I)
+C         END DO
+C         CALL TRANSPOSE(3,DROT,TDR)
+C         CALL PRDMAT(N,DROT,TDR,TI)
+C         WRITE(MPI,*)'WINGET DROT'
+C         DO I=1,N
+C            WRITE(MPI,101)(DROT(I,J),J=1,N)
+C         END DO
+C         WRITE(MPI,*)'WINGET TI'
+C         DO I=1,N
+C            WRITE(MPI,101)(TI(I,J),J=1,N)
+C         END DO
+C     END IF
```

Comment:
Do our code style practices not apply to old Fortran?

Reply:
Not sure if this is a joke...